### DISCUSSION TITLE
Usability improvements, CLI and config

### DISCUSSION THREAD

**Discussion by mzabaluev** [movementlabsxyz/movement#381] [Ideas]
Things to do:

* #382
* #384
* #385
* #386
* #387

**Comment by l-monninger** [comment]
> a more abstract system with node config possibly coming from a networked key-value space or similar.

I believe we want this, but simplicity is paramount as well.

**Reply by l-monninger** [reply]
@mzabaluev Let's open up a separate discussion for this.

**Reply by mzabaluev** [reply]
Spun off as #382.

**Comment by l-monninger** [comment]
> Consistent, concise naming in configuration files.

@mzabaluev Let's go through all of our config and state files in a separate discussion.

**Reply by mzabaluev** [reply]
Spun off as #384.

**Comment by l-monninger** [comment]
> Review CLI commands and option names for consistency and usability.

@mzabaluev  I think there are separate discussions for each binary we run here. Can you open them up?

**Reply by mzabaluev** [reply]
What other user-facing binaries should we consider?
The node binary is one: we can convert most of the settings obtained from environment variables to CLI options.
What else? Faucet?

**Comment by l-monninger** [comment]
> MVP: support local file (separate from config) and AWS Secrets (or KMS?).

I think there should be a generalized backend layer. Again, a separate discussion.

**Reply by mzabaluev** [reply]
This is now #387.

**Comment by l-monninger** [comment]
> Consider maintaining the CLI "front" binary as a crate hosted in movement.

I actually think a lot of this gets simpler with #340 . Maybe we can recommend an associated clap-compatible pattern?

**Reply by mzabaluev** [reply]
That's purely for the CLI, different from actors to organize async tasks in the node.

**Reply by mzabaluev** [reply]
Started #386 specifically about the needs of movement CLI.
